### ISSUE TITLE
Create Link Fetcher Lambda, To Download Queue, and Scheduler

### ISSUE THREAD

**Issue by ciaransweet** [NASA-IMPACT/hls-sentinel2-downloader-serverless#4]
# TL;DR

The Lambda function which queries IntHub needs to be created so that it:

* Runs on a schedule (perhaps once a day)
* Queries IntHub for all lints within the past 21 days
* If a link doesn't already exist in the DB, add it to `granules` and update `granule_count`
* Adds the new links to the `To Download` Queue

![image](https://user-images.githubusercontent.com/9111975/102629816-94344580-4143-11eb-9c7f-ffd2c9693d92.png)

# Info

Currently, link fetching is performed every 4 hours within `hls-s2-downloader`.

The pseudo code for the current link fetching process is:

```python
get_todays_date()
for day in days_between_today_and_21_days_ago:
    days_available_links_count, days_fetched_links_count = get_available_and_fetched_links(day)
    query = generate_query_for_day(day, start_point=days_fetched_links_count)
    
    while True: # We query till there are no more results available
        results = get_query_results(query)
        update_granule_count_with_available_links(results['total_results'])
        
        for result in results:
            if result['tile_id'] in tiles_to_ignore:
                continue
            create_granule_in_table(result)
            add_granule_to_download_queue(result)
    
    query['start'] += len(results)
    update_status_table()
    update_granule_count_table()
```

Refer to [this](https://github.com/NASA-IMPACT/hls-sentinel2-downloader/blob/version2-ajinkya/src/main.py#L57-L63) to follow the full flow, checks for exiting the loop early etc have been omitted from the pseudo code

An example of scheduling a lambda based on time can be found [here](https://github.com/developmentseed/ss-spire-ingestion-kickoff/blob/main/app/app_stack.py#L50-L61)

I think the SQS queue should enforce non-duplicate messages by default on deployment, so as to ensure we don't re-ingest data needlessly

# Acceptance criteria

* Lambda function and associated methods within are unit tested
* SQS Queue is deployed
* Event is deployed on an agreed schedule
* Integration test is run which ensures:
    * Links are fetched (This can be restricted for testing purposes by mocking the API with [API Gateway Mock Integrations](https://docs.aws.amazon.com/apigateway/latest/developerguide/how-to-mock-integration.html)
    * The `granule_count` table is correctly updated
    * The `granule` table is correctly updated
    * The SQS queue has the correct messages
